docs/tasks/administer-cluster/coredns.md

```diff
@@ -20,15 +20,34 @@ discovery.
 
 ## Installing CoreDNS with kubeadm
 
-In Kubernetes 1.9, [CoreDNS](https://coredns.io) is available as an alpha feature and
-may be installed by setting the `CoreDNS` feature gate to `true` during `kubeadm init`:
+In Kubernetes 1.9, [CoreDNS](https://coredns.io) is available as an alpha feature, and
+in Kubernetes 1.10 it is available as a beta feature. In either case, you may install
+it during cluster creation by setting the `CoreDNS` feature gate to `true` during `kubeadm init`:
 
 ```
 kubeadm init --feature-gates=CoreDNS=true
 ```
 
 This installs CoreDNS instead of kube-dns.
 
+## Upgrading an Existing Cluster with kubeadm
+
+In Kuberentes 1.10, you can also move to CoreDNS when you use `kubeadm` to upgrade
+a cluster that is using `kube-dns`. In this case, `kubeadm` will generate the CoreDNS configuration
+("Corefile") based upon the `kube-dns` ConfigMap, preserving configurations for federation,
+stub domains, and upstream name server.
+
+Note that if you are running CoreDNS already in your cluster, this conversion is not possible,
+as your Corefile may contain other customizations which should not be overwritten. In that
+case, it is recommended that you update your Corefile according to the changes you can find
+in the [recommended CoreDNS Kubernetes manifest](https://github.com/coredns/deployment/tree/master/kubernetes).
+
+In particular, changes that may be important are:
+
+* Updating the image to `coredns/coredns:1.0.6`
+* Changing `upstream /etc/resolv.conf` to just `upstream`, which changes external service CNAME lookups
```

Comment:
This change is done automatically right now as the CoreDNS configmap goes back to its default configuration after upgrade. 
We should ask the user to save their corefile/configmap before they upgrade to avoid them losing their own configurations.

Reply:
@rajansandeep are you okay with me assigning you to be the technical reviewer?  You seem to already have earned that role. :-)

Reply:
@bradtopol yes sure! I don't have rights to approve and lgtm the PR though.